src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java

```diff
@@ -570,18 +567,25 @@ public void unsetWakeWord() {
   }
 
   @Override
-  public void setLocale(String code) {
-    locale = new Locale(code);
+  public String getLanguage() {
+    return Runtime.getInstance().getLanguage();
   }
 
   @Override
-  public String getLanguage() {
-    return locale.getLanguage();
+  public Locale getLocale() {
+    return Runtime.getInstance().getLocale();
   }
 
   @Override
-  public Locale getLocale() {
-    return locale;
+  public Map<String, Locale> getLocales() {
+    return Locale.getAvailableLanguages();
+  }
+
+  @Override
+  public void setLocale(String code) {
+    // locale = new Locale(code);
+    Runtime.getInstance().setLocale(code);

```

Comment:
Setting the locale on this service should not change locale for the platform.

Reply:
I believe it should - the details of hierarchy of locales over the hierarchy of properties for multiple locales - first we need to make it work as best as possible with a single Locale .. mutliple Locales can be addressed in future functionality

Reply:
There is no validation that you can set the locale of the speech recognizer that the speech recognizer supports it.  This is very misleading...